src/krylov_phiv.jl

```diff
@@ -153,30 +148,31 @@ function ExponentialUtilities.expv!(w::GPUArraysCore.AbstractGPUVector{Tw},
         F = eigen!(SymTridiagonal(cache))
         expHe = F.vectors * (exp.(lmul!(t, F.values)) .* @view(F.vectors[1, :]))
     else
-        lmul!(t, cache)
-        expH = cache
-        _exp!(expH)
+        # lmul!(t, cache)
+        # expH = cache
+        # _exp!(expH)
+        # expHe = @view(expH[:, 1])
+        expH = exponential!(t * cache, expmethod)
         expHe = @view(expH[:, 1])
     end
 
-    copyto!(dexpHe, expHe)
-    lmul!(beta, mul!(w, @view(V[:, 1:m]), dexpHe)) # exp(A) ≈ norm(b) * V * exp(H)e
```

Comment:
is this extra allocation necessary?

Reply:
It needs to be a GPUArray type to perform the `lmul!` multiplication. But expHe is a CPUArray, so I  needed to move it into the GPU.

Reply:
Why is `expHe` on the CPU? From 156 it should be a view of an array on the GPU?

Reply:
Yes, I followed the concept of the Hessenberg matrix for the arnoldi iteration, which is alway a CPU matrix. So, I think that, since the krylov dimension is usually low, it was better to mantain that matrix on the CPU.

https://github.com/SciML/ExponentialUtilities.jl/blob/861bea82381a27c1e47249e9b272b0a19c9a2618/src/arnoldi.jl#L219-L226

What do you think?